docs/proposals/remediation/cluster-status-based-remedy.md

```diff
@@ -0,0 +1,383 @@
+- If `detectionFlag` is false and this Service domain name resolution is successful, then record the current time, execute the next command, if it lasts for a period of time (configured through the proxy component startup parameter `--detect-failure-threshold`, the default is 30s) Service domain name resolution successes, then it is determined that the Service domain name resolution function is normal; 
+- If `detectionFlag` is false and this Service domain name resolution is failed, the Service domain name resolution function is determined to be abnormal;
+
+#### Reporting of detection results
+
+For detection of Service domain name resolution, the status condition of the Cluster object in the Karmada control plane will be refreshed after each detection.
+
+### Policy control
+
+A new independent Remedy API is added to provide scalable capabilities, allowing users to configure matching conditions, and the system triggers action execution based on cluster status conditions.
+
+By configuring Remedy resources, the system will automatically remove the traffic on the cluster when detecting an abnormality in the Service domain name resolution function of a cluster, that is, remove the backend service address in the faulty cluster from the LoadBalance Service to improve the global availability of the MultiClusterIngress.
+
+In addition, administrators can manually configure immediate policies that trigger targeted actions to be executed immediately.
+
+A Remedy consists of three elements: trigger conditions (cluster status condition), focus on subjects (cluster), and execute actions
+
+#### API Definition
+
+The new `Remedy` CRD is used to describe the action management strategy driven by cluster status. This resource is a Cluster-level resource. Its Group is `remedy.karmada.io` and its Version is `v1alpha1`.
+
+> Note: Remedy CRD is pre-installed during Karmada deployment.
+
+```go
+// Remedy represents the cluster-level management strategies based on ClusterState.
+type Remedy struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec represents the desired behavior of Remedy.
+	// +required
+	Spec RemedySpec `json:"spec"`
+}
+
+// RemedySpec represents the desired behavior of Remedy.
+type RemedySpec struct {
+    // ClusterAffinity specifies the clusters that Remedy needs to pay attention to.
+    // For clusters that meet the DecisionConditions, Actions will be preformed.
+    // If empty, all clusters will be selected.
+    // +optional
+    ClusterAffinity *ClusterAffinity `json:"clusterAffinity,omitempty"`
```

Comment:
Minor formatting flaws, indentation may need to be standardized a bit

Reply:
Thanks, It has been updated.